### DISCUSSION TITLE
custom taxonomy collection relationship

### DISCUSSION THREAD

**Discussion by paulhaem** [statamic/cms#4849] [Troubleshooting]
Hello,
I have different kind of categories organised in different kind of taxonomies. For clarity I named them the same as the collection (always 1) they relate to. Now inside the collection blueprint I would like to alter the label and handle of the taxonomy field to e.g. "Category" instead of "Blog". Now after saving and reloading Statamic shows the updated taxonomy field and added the default taxonomy field to the blueprint. Deleting the default taxonomy field and reloading always leads to the same result. Only after removing the collection from the taxonomy config I am able to delete the default field. But now I am not able to query the terms using the collection filter - of course.

Its working just fine and maybe this is by design and intended but I think one should be able to alter the label (for user clarification) and handle (for code readability). In addition it leads to quite confusing URLs e.g. `/blog/blog/{category}`. But `/blog/blog-category/{category}` is not any better. Due to multiple different _categories_ they need to stay unique. Glad for any thoughts on this!

**Comment by edalzell** [comment]
handles can be changed in the files, not via CP. After you make the change everywhere don't forget to clear the cache.

**Reply by paulhaem** [reply]
I deleted all collection entries. Altered `handle` and `display` in the `resources/blueprints/collections/{collection}/{collection}.yaml` file. Then I cleared the cache using the utilities and the artisan command. Now when I head back to the blueprint Statamic shows the altered field and the default taxonomy field. When I look into the before mentioned `{collection}.yaml` file there is no entry for the default taxonomy field. After saving though the entry is being added to the file.
